StringRep.hs

```diff
@@ -70,21 +70,31 @@ postProc :: [Name] -> Term -> Term
 postProc ctx (Lam (M b) t) = Lam (M b) $ postProc (b:ctx) t
 postProc ctx (Ap t1 t2) = Ap (postProc ctx t1) (postProc ctx t2) 
 postProc ctx (Const x) | Just v <- elemIndex x ctx = LocalVar v
-                        | otherwise = Const x
+                       | otherwise = Const x
 
 type Error = MS.MisoString
 
--- TODO: need to unfold the Ap back to App Term [Term] for pretty print if_then_else_, current pretty printer only works for binary operator
+printHoley :: MS.MisoString -> [B.Builder] -> B.Builder
+printHoley origin operends = goPrintHoley origin origin operends
+goPrintHoley origin op [] = B.fromText $ MS.fromMisoString op
+goPrintHoley origin op (x:xs) = case MS.uncons op of 
+  Just ('_', ops) | origin == op -> x <> " " <> goPrintHoley origin ops xs
+                  | ops == (MS.toMisoString ("" :: String)) -> " " <> x <> goPrintHoley origin ops xs
```

Comment:
with {-# LANGUAGE OverloadedStrings #-} you can just say `""` instead of `MS.toMisoString ("" :: String)`